spec/features/consumer/shopping/unit_price_spec.rb

```diff
@@ -0,0 +1,77 @@
+require 'spec_helper'
+
+feature "As a consumer, I want to check unit price information for a product ;", js: true do
+  include AuthenticationHelper
+  include WebHelper
+  include ShopWorkflow
+  include UIComponentHelper
+
+  let(:distributor) { create(:distributor_enterprise, with_payment_and_shipping: true) }
+  let(:supplier) { create(:supplier_enterprise) }
+  let!(:order_cycle) { create(:simple_order_cycle, distributors: [distributor], variants: [product.variants.first]) }
+  let(:product) { create(:simple_product, supplier: supplier) }
+  let(:variant) { product.variants.first }
+  let(:order) { create(:order, order_cycle: order_cycle, distributor: distributor) }
+  let(:exchange1) { order_cycle.exchanges.to_enterprises(distributor).outgoing.first }
+  let(:user) { create(:user, password: "password", password_confirmation: "password") }
+
+  before do
+    set_order order
+    exchange1.update_attribute :pickup_time, "monday"
+    add_variant_to_order_cycle(exchange1, variant)
+    allow(OpenFoodNetwork::FeatureToggle)
+      .to receive(:enabled?).with(:unit_price, anything) { true }
+  end
+
+  describe "for the shopfront" do
+    before do
+      visit shop_path
+    end
+    
+    it "shows the unit price information with the question mark icon." do
+      expect(page).to have_selector '.variant-unit-price'
+      within '.variant-unit-price' do 
+        expect(page).to have_selector '.question-mark-icon'
+      end
+    end
+
+    it "one click on the question mark icon should open the tooltip." do
+      expect(page).to have_selector '.variant-unit-price'
+      find('.question-mark-icon').click
+      expect(page).to have_selector '.joyride-tip-guide.question-mark-tooltip'
+      within '.joyride-tip-guide.question-mark-tooltip' do 
+        expect(page).to have_content I18n.t('js.shopfront.unit_price_tooltip')
+      end
+    end
+
+    it "opening the tooltip and then click outside should close the tooltip." do
+      find('.question-mark-icon').click
+      page.find("body").click
+      expect(page).not_to have_selector '.joyride-tip-guide.question-mark-tooltip'
+      expect(page).to have_no_content I18n.t('js.shopfront.unit_price_tooltip')
+    end
```

Comment:
Now I see that this test could be a follow-up of the one above. It relies on the final state of the previous test. I wonder what we should favor here: performance or readability. I guess renaming the test above as `allows opening and closing the tooltip` and merging the two it would still be readable and a bit more performant. Thoughts? @andrewpbrett @jibees @mkllnk @Matt-Yorkley ?

Reply:
I'm wondering if tests in the same `describe` have to be independent of each other? Good question, no strong opinion on it!

Reply:
it's just a convention https://relishapp.com/rspec/rspec-core/docs/example-groups/basic-structure-describe-it but to me, for this DSL to be useful, the `it`s should all refer to what is said in the `describe`. That is, the same subject.

Reply:
In feature specs I favour performance over readability. It's still readable. And it can be made into a user journey without much repetition. It saves a lot of time.

Reply:
so if I understood correctly @mkllnk also agrees to merge these two tests (this one and the one above) into a single one that opens and then closes the tooltip.

Reply:
This is done. Thanks to you both! 🙏

Reply:
sorry to be so picky but now the first test is pointless. This one wouldn't work without the first one.